cli/medperf/commands/profile.py

```diff
@@ -0,0 +1,117 @@
+import os
+import typer
+import configparser
+
+from medperf import config
+from medperf.decorators import docstring_parameter
+from medperf.utils import parse_context_args, dict_pretty_print, pretty_error
+
+app = typer.Typer()
+
+
+def read_config():
+    config_p = configparser.ConfigParser()
+    config_path = os.path.join(config.storage, config.config_path)
+    config_p.read(config_path)
+    return config_p
+
+
+def write_config(config_p: configparser.ConfigParser):
+    config_path = os.path.join(config.storage, config.config_path)
+    with open(config_path, "w") as f:
+        config_p.write(f)
+
+
+@app.command("create", context_settings={"allow_extra_args": True, "ignore_unknown_options": True})
+@docstring_parameter(" | ".join(config.customizable_params))
+def create(
+    ctx: typer.Context,
+    name: str = typer.Option(..., "--name", "-n", help="Profile's name"),
+):
+    """Creates a new profile for managing and customizing configuration
+
+    Extra arguments in the format "--key=value" will be handled as custom configuration
+    parameters that will be stored under the new profile
+
+    Arguments:
+
+    {0}
+    """
+    args = parse_context_args(ctx.args)
+    config_p = read_config()
+
+    filtered_params = {k:v for k, v in args.items() if k in config.customizable_params}
+    if name in config_p:
+        pretty_error("A profile with the same name already exists", config.ui)
+
+    config_p[name] = filtered_params
+    write_config(config_p)
+
+
+@app.command("set", context_settings={"allow_extra_args": True, "ignore_unknown_options": True})
+@docstring_parameter(" | ".join(config.customizable_params))
+def set(ctx: typer.Context):
+    """Assign key-value configuration pairs to the current profile.
+
+    Extra arguments in the format "--key=value" will be handled as custom configuration
+
+    Available arguments:
+
+    {0}
+    """
+    profile = config.profile
+    args = parse_context_args(ctx.args)
+    config_p = read_config()
+
+    filtered_params = {k:v for k, v in args.items() if k in config.customizable_params}
+    current_config = config_p[profile]
```

Comment:
I am concerned about this.
A small typo in the key name (e.g. timeout->timeuot) when configuring a profile would have unexpected consequences.

Reply:
You are right, instead of filtering, we should validate and throw errors if an invalid argument is passed. I'll implement this